bin/select_mock_targets

```diff
@@ -96,6 +96,26 @@ log.info('Reading configuration file {}'.format(args.config))
 with open(args.config, 'r') as pfile:
     params = yaml.safe_load(pfile)
 
+# Set up MASKBITS lookup. Logic check for the necessary files, if not there wget the per brick files for
+# all overlapping pixels. 
+if 'legacy_dir' in params.keys():    
+  # E.g. global/project/projectdirs/cosmo/data/legacysurvey/dr8/
+  # wget https://portal.nersc.gov/project/cosmo/data/legacysurvey/dr8/north/coadd/338/3380p295/    
+  log.info('Assuming maskbits & depths based on LEGACYDIR={}'.format(params['legacy_dir']))
+
+  # Check the survey bricks file exists.  We'll use this to ensure that the local files are compelte for
+  # the healpixels being requested.  
+  legacy_dir = params['legacy_dir']
+  allbricks  = legacy_dir + '/survey-bricks.fits.gz'
+
+  if not os.path.isfile(allbricks):
+      raise  RuntimeError('survey-bricks.fits.gz is required under legacy_dir.')
+  
+else:
+  params['legacy_dir'] = None
+    
+  log.info('Assuming simple homogeneous depths & no masking.')
+
```

Comment:
I don't think we can realistically expect people to copy over entire brick-sized `coadd` files to their laptop when all we need are the mean properties at the location of a finite set of random/mock sources.

Reply:
Yeh, it'd have been better had we been able to coordinate more at the start.  Both I and @geordie666  didn't see this as particularly useful, but we understood that to be the thing that was compatible with "being able to run on a laptop".  I.e. we tried to support also running/development the pre-burner on a laptop - by requiring only the minimal number of files given a list of heal pixels.  Maybe you're willing to give up on the pre-burner on a laptop, this wasn't clear to me before.  I certainly follow the logic of trying to separate the two steps and this probably wouldn't be hard.   

My one "concern", which I think you're remembering from the telecon, is that I think you have in mind populating PSF depths etc. for something like the mocks that currently exist, e.g. ~20, including dark sky, etc.  There is an additional use case of _just the mask_ for _~1000s_ of (approximate) mocks in the clustering analysis, for which it's less clear that we want redundant information saved to disk in this case.  I'd suggest it makes sense to develop select_mock_targets in such a way that it's a one stop shop for both exercises.  I.e. in minimal mode, it turns a (ra, dec, z) mock into something accepted by downstream pipeline code, but which has flags etc. that let you dial up the sophistication of effects included.  There's a discussion planned on this at the cosmosim group telecon on Thursday.